### PR TITLE
Fix scrolling to resolved comment anchors

### PR DESCRIPTION
**Pull request by zeripath**
As described on discord, when the window.location.hash refers to a
resolved comment then the scroll to functionality does not work.

This PR fixes this.

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

File: web_src/js/index.js [hunk 1]
```diff
@@ -1172,6 +1172,22 @@ async function initRepository() {
 }
 
 function initPullRequestReview() {
+  if (window.location.hash && window.location.hash !== '' && window.location.hash.startsWith('#issuecomment-')) {
```
Comment (silverwind) [inline web_src/js/index.js]:
```suggestion
  if (window.location.hash && window.location.hash.startsWith('#issuecomment-')) {
```

empty string is falsy

File: web_src/js/index.js [hunk 2]
```diff
@@ -1172,6 +1172,22 @@ async function initRepository() {
...
+    const commentDiv = $(`${window.location.hash}`);
+    if (commentDiv) {
+      // get the name of the parent id
+      const groupID = commentDiv.parents('div[id^="code-comments-"]').attr('id');
+      if (groupID && groupID.startsWith('code-comments-')) {
+        const id = groupID.substr(14);
```
Comment (silverwind) [inline web_src/js/index.js:1181]:
```suggestion
        const id = groupID.substring(14);
```

File: web_src/js/index.js [hunk 3]
```diff
@@ -1172,6 +1172,22 @@ async function initRepository() {
 }
 
 function initPullRequestReview() {
+  if (window.location.hash && window.location.hash !== '' && window.location.hash.startsWith('#issuecomment-')) {
+    const commentDiv = $(`${window.location.hash}`);
```
Comment (silverwind) [inline web_src/js/index.js]:
```suggestion
    const commentDiv = $(window.location.hash);
```

File: web_src/js/index.js [hunk 4]
```diff
@@ -1172,6 +1172,22 @@ async function initRepository() {
...
+    if (commentDiv) {
+      // get the name of the parent id
+      const groupID = commentDiv.parents('div[id^="code-comments-"]').attr('id');
```
Comment (silverwind) [inline web_src/js/index.js]:
```suggestion
      const groupID = commentDiv.closest('div[id^="code-comments-"]').attr('id');
```

**Review by silverwind — APPROVED** [review]
Did not test this but change seems benign enough to not cause issues.

**Review by techknowlogick — APPROVED** [review]

**Comment by techknowlogick** [comment]
please send backport :)

**Pull request merged**